### PR TITLE
Feature/login and register

### PR DESCRIPTION
**Pull request by illuminat3**
Added the ability to login and register accounts

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -107,3 +107,9 @@ yarn.lock
 .idea_tasks/
 .idea/workspace/
 .idea_description.iml
+
+#Stop accidental uploads of the database
+*.db
+
+#Stop efcore migrations uploads
+Migrations/
```
Comment (mbaksh01) [inline .gitignore:115]: No

File: FoodTracker/Pages/Login.razor.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+using FoodTracker.Service.DataServices.Abstraction;
+using FoodTracker.Service.DataServices.DataAccess.Abstraction;
+using FoodTracker.Service.FunctionalServices.Abstraction;
+using Microsoft.AspNetCore.Components;
+
+namespace FoodTracker.Pages;
+
+public partial class Login
+{
+    [Inject] private NavigationManager _navigationManager {get; set;}
```
Comment (mbaksh01) [inline FoodTracker/Pages/Login.razor.cs:14]:
```suggestion
    [Inject] private NavigationManager NavigationManager { get; set; }
```

File: FoodTracker/Pages/Login.razor.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+        if (user == null)
+        {
+            _error.IsActive = true;
+            _error.Message = "Invalid username or email.";
+            return;
+        }
+        
+        var password = _hashService.HashString(_loginModel.Password);
```
Comment (mbaksh01) [inline FoodTracker/Pages/Login.razor.cs:52]:
```suggestion
        var hasedPassword = _hashService.HashString(_loginModel.Password);
```

File: FoodTracker/Service/DataServices/DataAccess/UserDataAccess.cs [hunk 4]
```diff
@@ -1,42 +1,57 @@
...
+        }
+
+        public async Task<User?> Get(string userId)
+        {
+            return await _context.Users.FindAsync(userId);
+        }
+
+        public async Task<IEnumerable<User>> GetAll()
+        {
+            return await _context.Users.ToListAsync();
```
Comment (mbaksh01) [inline FoodTracker/Service/DataServices/DataAccess/UserDataAccess.cs:44]:
```suggestion
            return await _context.Users.AsNoTracking().ToListAsync();
```

File: FoodTracker/Service/DataServices/UserService.cs [hunk 5]
```diff
@@ -1,27 +1,89 @@
...
+            try
+            {
+                if (_user != null)
+                {
+                    return _user;
+                }
+                
+                var userJson = await _localStorage.GetItemAsStringAsync(UserKey);
+                
+                if (string.IsNullOrEmpty(userJson))
```
Comment (mbaksh01) [inline FoodTracker/Service/DataServices/UserService.cs:33]:
```suggestion
                if (string.IsNullOrWhiteSpace(userJson))
```

File: README.md [hunk 6]
```diff
@@ -8,6 +8,27 @@ temporarily hosted on
...
+To do this use the following commands
+
+```bat
+dotnet tool install --global dotnet-ef
+```
+
+Make sure you are in the FoodTracker directory and run the follwing
+
+``` bat
+dotnet ef migrations add InitialCreate
```
Comment (mbaksh01) [inline README.md:23]: No

File: .gitignore (hunk 1 above)
Comment (illuminat3) [reply .gitignore:115]: >:(

File: FoodTracker/Pages/Login.razor.cs (hunk 3 above)
Comment (illuminat3) [reply FoodTracker/Pages/Login.razor.cs]: You had typo to manually resolved

File: README.md (hunk 6 above)
Comment (illuminat3) [reply README.md:23]: ? whats wrong with this

**Review by mbaksh01 — APPROVED** [review]

**Pull request merged**